pkg/resources/user.go

```diff
@@ -133,9 +137,6 @@ var userSchema = map[string]*schema.Schema{
 		Sensitive:   true,
 		Description: "Last name of the user.",
 	},
-	"tag": tagReferenceSchema,
```

Comment:
so why are tags removed from here, but they still are present e.g. in `CreateUserOptions` (and set in tests etc)? Just want to confirm: is it because we differentiate between sdk capabilities, but from the terraform resource side we would like to use tag_association resource to set these tags (in this case)?

Reply:
Yes, this is something  we should talk about, probably @scottwinkler would be really helpful in this conversation.
We have a separate `tag_association` resource which can be used for setting tags, and from what I could gather we want to move towards that approach. I based my implementation on recent warehouse sdk rewrite, which also removed the `tag` attribute from the schema.
One obvious downside of just removing "tag" attribute from the schema is that it breaks compatibility which I guess we want to avoid. Rewriting `tagReferenceSchema` to using new sdk implementation seems like quite a bit of work, but maybe it's still worth it

Reply:
we cannot introduce any more breaking changes to the provider prior to the next major release. We have to keep tags in, if it was already supported by the resource. I think we should support both ways of tagging resources. So perhaps deprecating this attribute was the wrong idea.

Reply:
Now that I look at it, I'm not sure if tags were actually supported for users resource. I might be mistaken, but it seems they were part of the schema, but we didn't do anything with them. In resources like `schema` we make use of the "tag" attribute and perform corresponding SQL queries, but for users and warehouse(on which I based this implementation) resources didn't really support tags as part of the resource.
So I think maybe we could just leave the tag schema as it was and try to add actual support for tags for the resources like users and warehouses(in a separate PR though, this one is already taking too long).